flink-libraries/flink-table/src/main/scala/org/apache/flink/table/functions/utils/TableSqlFunction.scala

```diff
@@ -40,17 +40,14 @@ class TableSqlFunction(
     name: String,
     udtf: TableFunction[_],
     rowTypeInfo: TypeInformation[_],
-    returnTypeInference: SqlReturnTypeInference,
-    operandTypeInference: SqlOperandTypeInference,
-    operandTypeChecker: SqlOperandTypeChecker,
-    paramTypes: util.List[RelDataType],
+    typeFactory: FlinkTypeFactory,
     functionImpl: FlinkTableFunctionImpl[_])
   extends SqlUserDefinedTableFunction(
     new SqlIdentifier(name, SqlParserPos.ZERO),
-    returnTypeInference,
-    operandTypeInference,
-    operandTypeChecker,
-    paramTypes,
+    ReturnTypes.CURSOR,
+    createOperandTypeInference(name, udtf, typeFactory),
+    createOperandTypeChecker(name, udtf),
+    null,
```

Comment:
Is it safe to pass `null` here? Couldn't we create this parameter from `rowTypeInfo`?

Reply:
I think it is fine. `ScalarSqlFunction` and `AggregateSqlFunction` also pass `null` and works good.